Update cross-origin isolation guide message (#4908)
GoogleChrome/web.dev: src/site/content/en/secure/cross-origin-isolation-guide/index.md

---

This guide shows you how to enable cross-origin isolation. Cross-origin
isolation is required if you want to use
[SharedArrayBuffer](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer),
[`performance.measureUserAgentSpecificMemory()`](/monitor-total-page-memory-usage/),
or the JS Self-Profiling API.

{% Aside 'objective' %}
If you landed this page by a SharedArrayBuffer deprecation message, it's likely
either your website or one of the resources you emebed to your website is using
SharedArrayBuffer. When Chrome hits 91, the functionality that has been

> available through SharedArrayBuffer will no longer work. take one of the

Reviewer: It still works, but only in COI context.

> available through SharedArrayBuffer will no longer work. take one of the

Reviewer: [...] take -> Take

following actions to fix the issue in advance.

* Turn on the cross-origin isolation to keep using SharedArrayBuffer.

> * Notify the third-party that actually uses SharedArrayBuffer to take action.

Reviewer: Notify the third-party provider/vendor(?)

{% endAside %}

## Determine where in your website a SharedArrayBuffer is used

If you are not sure where in your site a SharedArrayBuffer is used, there are
two ways:

> * Using Chrome DevTools

Reviewer: Shall we mark Chrome DevTools as 'Basic' and Reporting API as 'Advanced'?

---

This guide shows you how to enable cross-origin isolation. Cross-origin
isolation is required if you want to use
[SharedArrayBuffer](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer),
[`performance.measureUserAgentSpecificMemory()`](/monitor-total-page-memory-usage/),
or the JS Self-Profiling API.

> {% Aside 'objective' %}
> If you landed this page by a SharedArrayBuffer deprecation message, it's likely
> either your website or one of the resources you emebed to your website is using
> SharedArrayBuffer. When Chrome hits 91, the functionality that has been
> available through SharedArrayBuffer will no longer work without cross-origin
> isolation. Take one of the following actions to fix the issue in advance.
> 
> * Turn on the cross-origin isolation to keep using SharedArrayBuffer.
> * Notify the third-party vendor/provider that actually uses SharedArrayBuffer to
>   take action.
> {% endAside %}

Reviewer 2: ```suggestion
{% Banner 'info', 'body' %}
Starting in Chrome 91, functionalities that use `SharedArrayBuffer` will no longer 
work without cross-origin isolation. If you landed on this page due to a 
`SharedArrayBuffer` deprecation message, it's likely either your website or one of 
the resources embedded on it is using `SharedArrayBuffer`. To ensure nothing breaks 
on your website due to deprecation, follow the steps in this post.
{% endBanner %}

{% Aside 'objective' %}
* Turn on cross-origin isolation to keep using `SharedArrayBuffer`.
* If you rely on third-party code that uses `SharedArrayBuffer`, notify the third-party 
  provider to take action.
{% endAside %}
```

> ## Determine where in your website a SharedArrayBuffer is used
> 
> If you are not sure where in your site a SharedArrayBuffer is used, there are
> two ways:
> 
> * [Basic] Using Chrome DevTools
> * [Advanced] Using Deprecation Reporting
> 
> Skip to [Do an impact analysis](:analysis) if you already know where you are
> using SharedArrayBuffer.

Reviewer 2: ```suggestion
## Determine where in your website `SharedArrayBuffer` is used

If you are not sure where in your site a `SharedArrayBuffer` is used, there are
two ways find out:

* Using Chrome DevTools
* (Advanced) Using Deprecation Reporting

If you already know where you are using `SharedArrayBuffer`, skip to 
[Do an impact analysis](:analysis).
```

> ### [Basic] Using Chrome DevTools
> 
> Chrome has a tool for developers to inspect the website. You can identify the
> issue using it.
> 
> 1. [Open the Chrome
>    DevTools](https://developers.google.com/web/tools/chrome-devtools/open) on
>    the page you suspect it might be using a SharedArrayBuffer.
> 2. Select the **Console** panel.
> 3. The following message will show up if SharedArrayBuffer is used:
>       ```text
>       [Deprecation] SharedArrayBuffer will require cross-origin isolation as of M91, around May 2021. See https://developer.chrome.com/blog/enabling-shared-array-buffer/ for more details. common-bundle.js:535
>       ```
> 4. The filename and the line number indicates where the SharedArrayBuffer comes
>    from. If it's a third-party library, please contact the developer to fix the
>    issue. If it's implemented as part of your website, follow the guide below to
>    enable cross-origin isolation.

Reviewer 2: ```suggestion
### Using Chrome DevTools

[Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/open)
allows developers to inspect websites. 

1. [Open the Chrome
   DevTools](https://developers.google.com/web/tools/chrome-devtools/open) on
   the page you suspect might be using `SharedArrayBuffer`.
2. Select the **Console** panel.
3. If the page is using `SharedArrayBuffer`, the following message will show up:
      ```text
      [Deprecation] SharedArrayBuffer will require cross-origin isolation as of M91, around May 2021. See https://developer.chrome.com/blog/enabling-shared-array-buffer/ for more details. common-bundle.js:535
      ```
4. The filename and the line number at the end of the message (for example, `common-bundle.js:535`) 
   indicate where the `SharedArrayBuffer` is coming from. If it's a third-party library, 
   contact the developer to fix the issue. If it's implemented as part of your website, follow 
   the guide below to enable cross-origin isolation.
```

<figure class="w-figure">

> {% Img src="image/YLflGBAPWecgtKJLqCJHSzHqe2J2/GOgkyjAabePTc8AG22F7.png", alt="DevToools Console warning when SharedArrayBuffer is used without cross-origin isolation", width="800", height="163" %}

Reviewer 2: ```suggestion
{% Img src="image/YLflGBAPWecgtKJLqCJHSzHqe2J2/GOgkyjAabePTc8AG22F7.png", alt="DevToools Console warning when SharedArrayBuffer is used without cross-origin isolation", width="800", height="163", class="w-screenshot" %}
```

   <figcaption>

>       DevToools Console warning when SharedArrayBuffer is used without cross-origin isolation

Reviewer 2: ```suggestion
      DevToools Console warning when `SharedArrayBuffer` is used without cross-origin isolation.
```

   </figcaption>
</figure>

> ### [Advanced] Using Deprecation Reporting

Reviewer 2: ```suggestion
### (Advanced) Using Deprecation Reporting
```

> Some browsers have [a reporting functionality of deprecating
> APIs](https://wicg.github.io/deprecation-reporting/) to a specified endpoint.
> 
> 1. [Set up a deprecation report server and get the reporting
>    URL](/coop-coep/#set-up-reporting-endpoint). You can achieve this by either
>    finding a public service or building one by yourself.
> 2. Using the URL, set the following HTTP header to pages that are potentially
>    serving SharedArrayBuffer usage.
>       ```http
>       Report-To: {"group":"default","max_age":86400,"endpoints":[{"url":"THE_DEPRECATION_ENDPOINT_URL"}]}
>       ```
> 3. Once the header starts to propagate, the endpoint you registered to should
>    start collecting deprecation reports.
> 
> See an example implementation here:
> [https://first-party-test.glitch.me](https://first-party-test.glitch.me)

Reviewer 2: ```suggestion
Some browsers have [a reporting functionality of deprecating
APIs](https://wicg.github.io/deprecation-reporting/) to a specified endpoint.

1. [Set up a deprecation report server and get the reporting
   URL](/coop-coep/#set-up-reporting-endpoint). You can achieve this by either
   using a public service or building one yourself.
2. Using the URL, set the following HTTP header to pages that are potentially
   serving `SharedArrayBuffer`.
      ```http
      Report-To: {"group":"default","max_age":86400,"endpoints":[{"url":"THE_DEPRECATION_ENDPOINT_URL"}]}
      ```
3. Once the header starts to propagate, the endpoint you registered to should
   start collecting deprecation reports.

See an example implementation here:
[https://first-party-test.glitch.me](https://first-party-test.glitch.me).
```

> ## Do an impact analysis {: #analysis}

Reviewer 3: I've found this headline a bit confusing. Suggestion: 'Analyse the impact of cross-origin isolation'

> ## Do an impact analysis {: #analysis}

Reviewer 2: ```suggestion
## Analyze the impact of cross-origin isolation  {: #analysis}
```

---

This guide shows you how to enable cross-origin isolation. Cross-origin
isolation is required if you want to use

> [SharedArrayBuffer](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer),

Reviewer 2: ```suggestion
[`SharedArrayBuffer`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer),
```

allows developers to inspect websites. 

1. [Open the Chrome
   DevTools](https://developers.google.com/web/tools/chrome-devtools/open) on
   the page you suspect might be using `SharedArrayBuffer`.
2. Select the **Console** panel.
3. If the page is using `SharedArrayBuffer`, the following message will show up:
      ```text
      [Deprecation] SharedArrayBuffer will require cross-origin isolation as of M91, around May 2021. See https://developer.chrome.com/blog/enabling-shared-array-buffer/ for more details. common-bundle.js:535
      ```
4. The filename and the line number at the end of the message (for example, `common-bundle.js:535`) 
   indicate where the `SharedArrayBuffer` is coming from. If it's a third-party library, 
   contact the developer to fix the issue. If it's implemented as part of your website, follow 
   the guide below to enable cross-origin isolation.

<figure class="w-figure">
{% Img src="image/YLflGBAPWecgtKJLqCJHSzHqe2J2/GOgkyjAabePTc8AG22F7.png", alt="DevToools Console warning when SharedArrayBuffer is used without cross-origin isolation", width="800", height="163", class="w-screenshot" %}
   <figcaption>

>       DevToools Console warning when `SharedArrayBuffer` is used without cross-origin isolation.

Reviewer 2: ```suggestion
      DevToools Console warning when SharedArrayBuffer is used without cross-origin isolation.
```
